docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,543 @@
+# Implementing a Distribution (developer guide)
+
+This guide provides an overview on how to implement a distribution for version 4
+of PyMC3. It is designed for developers who wish to add a new distribution to
+the library. Users will not be aware of all this complexity and should instead
+make use of helper methods such as (TODO).
+
+PyMC3 {`Distributions`}`pymc3.distributions.Distribution` build on top of
+Aesara's {`RandomVariables`}`aesara.tensor.random.op.RandomVariable`, and
```

Comment:
I now realize I did not share any reference on how intersphinx works. The `{class}` thingy indicates that we are linking to an object of type class. intersphinx docs are quite bad so I tried to document everything and add examples at https://arviz-devs.github.io/arviz/contributing/developer_guide.html#reference-external-libraries. It is still missing the custom text option though:

```
{class}`pymc3.distributions.Distribution`                 -> pymc3.distributions.Distribution
{class}`~pymc3.distribution.Distribution`                 -> Distribution
{class}`Distributions <pymc3.distribution.Distribution>`  -> Distributions
```